File: CenyWPolsce.Java/CenyWPolsce/app/src/main/java/com/example/cenywpolsce/models/PriceChange.java
package com.example.cenywpolsce.models;

public class PriceChange {
    public PriceChange()
    {

    }

    public PriceChange(String id, String nazwa, double startPrice, double endPrice)
    {
        ID = id;
        Nazwa = nazwa;
        StartPrice = startPrice;
        EndPrice = endPrice;
    }

    public String ID;
    public String Nazwa;
    public double StartPrice;
    public double EndPrice;

    private double _change;
    private boolean _changeCalculated = false;
    public double Change() {
        if (_changeCalculated == false)
        {
            _change = ((EndPrice - StartPrice) / StartPrice) * 100;
            _changeCalculated = true;
        }

        return _change;
    }
}

File: CenyWPolsce.Blazor/Models/PriceChange.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ceny.Models
{
    public class PriceChange
    {
        public PriceChange()
        {

        }

        public PriceChange(string id, string nazwa, double startPrice, double endPrice)
        {
            ID = id;
            Nazwa = nazwa;
            StartPrice = startPrice;
            EndPrice = endPrice;
        }

        public string ID { get; set; }
        public string Nazwa { get; set; }
        public double StartPrice { get; set; }
        public double EndPrice { get; set; }

        private double _change;
        private bool _changeCalculated = false;
        public double Change
        {
            get
            { 
                if (_changeCalculated is false)
                {
                    _change = ((EndPrice - StartPrice) / StartPrice) * 100;
                    _changeCalculated = true;
                }
                    
                return _change;
            }
        }
    }
}
